src/client/CodeEditor/InteractiveWidgets/colorPicker.ts

```diff
@@ -8,39 +8,135 @@ export const colorPickerRegex = /#[0-9A-Fa-f]{6}/g;
 // Works without quotes to support CSS.
 export const colorPicker = (
   onInteract?: () => void,
-): InteractRule => ({
-  regexp: colorPickerRegex,
-  cursor: 'pointer',
-  onClick(
-    text: string,
-    setText: (newText: string) => void,
-  ) {
-    const startingColor: string = text;
-
-    const sel: HTMLInputElement =
-      document.createElement('input');
-    sel.type = 'color';
-    sel.value = startingColor.toLowerCase();
-
-    // `valueIsUpper` maintains the style of the user's code.
-    // It keeps the case of a-f the same case as the original.
-    const valueIsUpper: boolean =
-      startingColor.toUpperCase() === startingColor;
-
-    const updateHex = (e: Event) => {
-      const el: HTMLInputElement =
-        e.target as HTMLInputElement;
-      if (el.value) {
-        setText(
-          valueIsUpper ? el.value.toUpperCase() : el.value,
-        );
-      }
-      if (onInteract) onInteract();
-    };
-    sel.addEventListener('input', updateHex);
-    sel.click();
-  },
-});
+):  InteractRule => {
+  // Create the color picker element when the page loads
+  const sel: HTMLInputElement = document.createElement('input');
+  sel.type = 'color';
+  sel.style.position = 'fixed';
+  sel.style.left = '-9999px'; // Initially off-screen
+  sel.style.top = '-9999px';
+  document.body.appendChild(sel);
+
+  // Create a close button element to hide the color picker
+  const closeButton: HTMLButtonElement = document.createElement('button');
```

Comment:
Nice. There could be some additional work here to make it more similar to the "X" buttons in other parts of the UI, like closing tabs, but this is awesome as a first version. Excellent work! Thank you so much.

Reply:
I was testing this more and there seems to be a regression actually, where the text is not getting updated.

I'm considering temporarily reverting these changes so the color picker edits the text again.

Thanks for your work on this!

Reply:
Fixed in https://github.com/vizhub-core/vzcode/pull/884